File: backend/src/main/kotlin/ru/itmo/db/coursework/api/mapping/CircuitBoardDtoMapper.kt
package ru.itmo.db.coursework.api.mapping

import org.springframework.stereotype.Service
import ru.itmo.db.coursework.api.dto.CircuitBoardDto
import ru.itmo.db.coursework.model.CircuitBoard

@Service
class CircuitBoardDtoMapper {
    fun fromDto(dto: CircuitBoardDto) = with(dto, {
        CircuitBoard(
            id = id!!,
            name = name!!,
            sellPrice = sellPrice!!
        )
    })

    fun toDto(model: CircuitBoard) = with(model, {
        CircuitBoardDto(
            id = id,
            name = name,
            sellPrice = sellPrice
        )
    })
}

File: frontend/src/api/mapping/CircuitBoardDtoMapper.ts
import UserDto from "$src/api/dto/UserDto";
import User from "$src/game/model/User";
import CircuitBoard from "$src/game/model/CircuitBoard";
import CircuitBoardDto from "$src/api/dto/CircuitBoardDto";

export default class CircuitBoardDtoMapper {
  public static fromDto(dto: CircuitBoardDto) {
    return new CircuitBoard(dto.id, dto.name, dto.sell_price)
  }

  public static toDto(model: CircuitBoard) {
    return new CircuitBoardDto(model.id, model.name, model.sellPrice)
  }
}